pkg/cmd/pulumi/new.go

```diff
@@ -390,6 +420,195 @@ func useSpecifiedDir(dir string) (string, error) {
 	return cwd, nil
 }
 
+type AIPulumiUserInfo struct {
+	UserId string `json:"userId"`
+}
+
+type AIPromptRequestBody struct {
+	Language       string `json:"language"`
+	Instructions   string `json:"instructions"`
+	Model          string `json:"model"`
+	ResponseMode   string `json:"responseMode"`
+	ConversationID string `json:"conversationId"`
+	ConnectionID   string `json:"connectionId"`
+}
+
+// Iteratively prompt the user for input, sending their input as a prompt tp Pulumi AI
+// Stream the response back to the console, and repeat until the user is done.
+func runAINew(
+	ctx context.Context,
+	args newArgs,
+	opts display.Options,
+	prompt string,
+	language string,
+	yes bool,
+) (conversationURL string, err error) {
+	// Try to read the current project
+	project, _, err := readProject()
+	if err != nil && !errors.Is(err, workspace.ErrProjectNotFound) {
+		return "", err
+	}
+
+	b, err := currentBackend(ctx, project, opts)
+	if err != nil {
+		return "", err
+	}
+
+	name, _, _, err := b.CurrentUser()
+	if err != nil {
+		return "", err
+	}
+	languageOptions := []string{
+		"TypeScript",
+		"JavaScript",
+		"Python",
+		"Go",
+		"C#",
+	}
+	if language == "" {
+		if err = survey.AskOne(&survey.Select{
+			Message: "Please select a language for your project:",
+			Options: languageOptions,
+		}, &language, surveyIcons(opts.Color)); err != nil {
+			return "", err
+		}
+	}
+	var continuePrompt string
+	for continuePrompt, conversationURL, _, err = runAINewPromptStep(
+		opts,
+		language,
+		prompt,
+		yes,
+		name,
+		"",
+	); continuePrompt == "continue"; continuePrompt, conversationURL, _, err = runAINewPromptStep(
+		opts,
+		language,
+		prompt,
+		yes,
+		name,
+		continuePrompt,
+	) {
```

Comment:
I think there's a bug here where the connection ID isn't retrieved and passed to the subsequent calls. 

This `for expr; check; expr; { }` loop is a bit strange to me, too, hard to parse. Is this essentially a "do while"?

Reply:
> This for expr; check; expr; { } loop is a bit strange to me, too, hard to parse. Is this essentially a "do while"?

That's correct - there aren't many ergonomic ways to express that kind of control flow in golang, and this was about as clean as I could find.

It's definitely possible the connectionID isn't being passed through correctly, so let me take a look (though I do believe I tested this - never hurts to verify).

Reply:
Ah - I see the issue re: connectionID. Will fix this now.